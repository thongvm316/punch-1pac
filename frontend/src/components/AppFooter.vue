<template>
  <footer class="footer">
    <div class="container grid-xl">
      <div class="footer-items">
        <ul class="list-inline">
          <li><img src="../assets/logo.png"></li>
          <li><a :href="`${meta.base_url}/terms`">{{ $t('footer.terms') }}</a></li>
          <li><a :href="`${meta.base_url}/privacy`">{{ $t('footer.privacy') }}</a></li>
          <li><a :href="`${meta.base_url}/help`">{{ $t('footer.help') }}</a></li>
          <li><a :href="`${meta.base_url}/contact`">{{ $t('footer.contact') }}</a></li>
          <li><a :href="`${meta.base_url}/about`">{{ $t('footer.about') }}</a></li>
        </ul>
        <span>Â© 2018 1PAC Vietnam Co.,Ltd</span>
      </div>
    </div>
  </footer>
</template>

<script>
import { mapState } from 'vuex'

export default {
  name: 'app-footer',

  computed: {
    ...mapState('initialStates', [
      'meta'
    ])
  }
}
</script>
