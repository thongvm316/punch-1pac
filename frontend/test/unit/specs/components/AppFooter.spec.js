import { shallowMount } from '@vue/test-utils'
import wrapperOps from '../../supports/wrapper'
import setComputed from '../../supports/set-computed'
import initialStatesData from '../../supports/fixtures/initial-states.api'
import AppFooter from '@/components/AppFooter'

const wrapper = shallowMount(AppFooter, wrapperOps)

describe('when AppFooter was mounted', () => {
  it('should render correctly', () => {
    setComputed(wrapper, {
      meta: initialStatesData.meta
    })

    expect(wrapper.exists()).toBeTruthy()
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })
})
