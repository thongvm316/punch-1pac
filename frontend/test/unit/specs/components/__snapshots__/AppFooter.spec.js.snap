// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when AppFooter was mounted should render correctly 1`] = `
<footer class="footer">
  <div class="container grid-xl">
    <div class="footer-items">
      <ul class="list-inline">
        <li><img src="../assets/logo.png"></li>
        <li><a href="http://localhost:3000/terms">Terms</a></li>
        <li><a href="http://localhost:3000/privacy">Privacy</a></li>
        <li><a href="http://localhost:3000/help">Help</a></li>
        <li><a href="http://localhost:3000/contact">Contact us</a></li>
        <li><a href="http://localhost:3000/about">About</a></li>
      </ul> <span>Â© 2018 1PAC Vietnam Co.,Ltd</span>
    </div>
  </div>
</footer>
`;
